core/controllers/question_editor.py

```diff
@@ -0,0 +1,128 @@
+
+        question = question_services.get_question_by_id(question_id)
+
+        if question is None:
+            raise self.PageNotFoundException(
+                'The question with the given id doesn\'t exist.')
+
+        self.values.update({
+            'question_id': question.to_dict(),
+            'can_edit': question_services.check_can_edit_question(
+                self.user.user_id, question.question_id)
+        })
+
+        self.render_template(
+            'pages/question_editor/question_editor.html',
+            redirect_url_on_logout='/')
+
+
+class EditableQuestionDataHandler(base.BaseHandler):
+    """A data handler for questions which supports writing."""
+
+    @acl_decorators.can_view_any_question_editor
+    def get(self, question_id):
+        """Gets the data for the question overview page."""
+        # 'apply_draft' and 'v'(version) are optional parameters because the
+        # question history tab also uses this handler, and these parameters
+        # are not used by that tab.
+        version = self.request.get('v', default_value=None)
+        apply_draft = self.request.get('apply_draft', default_value=False)
+
+        question_data = question_services.get_user_question_data(
+            self.user_id, question_id, apply_draft=apply_draft,
+            version=version)
+
+        self.values.update(question_data)
+        self.render_json(self.values)
+
+    @acl_decorators.can_edit_question
+    def put(self, question_id):
+        """Updates properties of the given question."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        question = question_services.get_question_by_id(question_id)
+        if question is None:
+            raise self.PageNotFoundException(
+                'The question with the given id doesn\'t exist.')
+
+        version = self.payload.get('version')
+        self._require_valid_version(version, question.version)
+
+        commit_message = self.payload.get('commit_message')
+        change_dicts = self.payload.get('change_dicts')
+        change_list = []
+        for change in change_dicts:
+            change_list.append(
+                question_domain.QuestionChange(change))
+        try:
+            question_services.update_question_pages(
+                self.user_id, question_id, change_list,
+                commit_message)
+        except utils.ValidationError as e:
+            raise self.InvalidInputException(e)
+
+        question_dict = question_services.get_question_by_id(
+            question_id).to_dict()
+
+        self.values.update({
+            'question': question_dict
+        })
+
+        self.render_json(self.values)
+
+    @acl_decorators.can_delete_question
+    def delete(self, question_id):
+        """Handles Delete requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        log_debug_string = '(%s) %s tried to delete question %s' % (
```

Comment:
I am not sure if we should log all instances where a question is deleted. Any reason why?

Reply:
I thought logging it would be good. Same as for explorations...

Reply:
Ok, I am not against it. Just not sure if it is necessary.